sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -787,8 +787,9 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
     return ReturnValue(size_t{1});
   case PI_EXT_INTEL_DEVICE_INFO_MAX_COMPUTE_QUEUE_INDICES:
     return ReturnValue(pi_int32{1});
+  case PI_DEVICE_INFO_MAX_NUM_SUB_GROUPS:
+    return ReturnValue(pi_uint32{1}); // Minimum required by SYCL 2020 spec
```

Comment:
Note for @intel/dpcpp-esimd-reviewers: I don't really know which exact value should be returned here or which value is valid in context of ESIMD programming model. Therefore, I went with a minimum required value from the spec, simply to make E2E test also pass on ESIMD emulator device (https://github.com/intel/llvm-test-suite/pull/1607)

Reply:
@dongkyunahn-intel - Would you advise here please? I think 1 is too conservative, but don't know how to reasonably limit the returned value.

Reply:
I'm not sure either. Basically, ESIMD_EMU is running on top of CM_EMU, and CM_EMU does not support sub-group.

Reply:
Value `1` effectively means that there can only be one sub-group per work-group, i.e. the whole work-group is treated as a one huge sub-group. So, to me it looks like it should be ok to return `1` when sub-groups aren't really supported

Reply:
@intel/dpcpp-esimd-reviewers, there seem to be no objections from @dongkyunahn-intel against returning `1`here. Could you please approve this patch then?